storage/redis/client.go

```diff
@@ -117,11 +117,16 @@ func (client *Client) GetAll(keys storage.Keys) (storage.Values, error) {
 
 	values := []storage.Value{}
 	for _, result := range results {
-		s, ok := result.(string)
-		if !ok {
-			return nil, Error.New("invalid result type %T", result)
+		if result == nil {
```

Comment:
It's unclear why this change is necessary here.

Reply:
I want to make sure I append the nil value to the results returned by GetAll and the original implementation will append an empty storage value, rather than actual nil.

Reply:
I don't think the underlying redis nor bolt implementations give that guarantee. This unfortunately means that everything above them should be able to deal with the `nil == storage.Value{}` situation.

Reply:
Although, the change here is correct for another reason. i.e. GetAll shouldn't fail on a missing value.

Reply:
It looks like they should return nil if the key is not found...
https://redis.io/commands/mget
https://godoc.org/github.com/boltdb/bolt#Bucket.Get

Reply:
What do they return for an empty value?

Reply:
empty string for key returns nil, "ERR wrong number of arguments for 'mget' command" if no keys are passed in. I will do a check for that, so the program doesn't panic.